sos/plugins/watchdog.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (C) 2018 Red Hat, Inc., Reid Wahl <nwahl@redhat.com>
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+
+from sos.plugins import Plugin, RedHatPlugin
+
+import re
+
+
+class Watchdog(Plugin, RedHatPlugin):
+    '''Watchdog information.'''
+    plugin_name = 'watchdog'
+    profiles = ('system',)
+    packages = ('watchdog',)
+    option_list = [
+        ('conf_file', 'watchdog config file', 'fast', '/etc/watchdog.conf'),
+    ]
+
+    def get_log_dir(self):
+        '''Get watchdog log directory.
+
+        Get watchdog log directory path configured in ``conf_file``.
+
+        :returns: The watchdog log directory path.
+        :returntype: str.
+        :raises: IOError if ``conf_file`` is not readable.
+        '''
+        log_dir = None
+        log_dir_pattern = re.compile(r'log-dir\s*=\s*(.*)')
+
+        with open(self.get_option('conf_file'), 'r') as conf_file:
+            for line in conf_file:
+                line = line.split('#')[0].strip()
+
+                try:
+                    match = log_dir_pattern.match(line)
```

Comment:
I'm still not really convinced that this warrants a regex, but hopefully we might see some progress on a general config parsing facility soon which would make this moot.

Reply:
What do you think of this? Still need to be able to strip whitespace due to the format of the file. `ValueError` for no `=` separator.

```
                try:
                    (key, value) = line.split('=', 1)
                    if key.strip() == 'log-dir':
                        log_dir = value.strip()
                except ValueError:
                    pass
```

Can then axe the `re` import since this was the only application of it.

Reply:
That's the kind of parsing I tend to favour, but we're really getting into personal preference at this point - there are plenty of plugins that do use `re` for this sort of task (and even more trivial tasks..).

I wouldn't hold a plugin up just for a minor change like this and in any case, we're actively looking at adding some kind of general config handling mechanism now anyway so even if it does go in using `re` to start it may not stay in that form for long.

Reply:
I like it better too now that I'm looking at it, for such simple parsing. Much easier to see what's actually going on.